test/server/listener_manager_impl_test.cc

```diff
@@ -141,22 +141,30 @@ class ListenerManagerImplWithRealFiltersTest : public ListenerManagerImplTest {
                                                                                     context);
             }));
     socket_ = std::make_unique<NiceMock<Network::MockConnectionSocket>>();
-    address_.reset(new Network::Address::Ipv4Instance("127.0.0.1", 1234));
+    local_address_.reset(new Network::Address::Ipv4Instance("127.0.0.1", 1234));
+    remote_address_.reset(new Network::Address::Ipv4Instance("127.0.0.1", 1234));
   }
 
   const Network::FilterChain*
   findFilterChain(uint16_t destination_port, bool expect_destination_port_match,
                   const std::string& destination_address, bool expect_destination_address_match,
                   const std::string& server_name, bool expect_server_name_match,
                   const std::string& transport_protocol, bool expect_transport_protocol_match,
-                  const std::vector<std::string>& application_protocols) {
-    const int times = expect_destination_port_match ? 2 : 1;
+                  const std::vector<std::string>& application_protocols,
+                  const std::string& source_address, bool expect_source_type_match) {
```

Comment:
Please add `bool expect_application_protocol_match` (meaning that there was `application_protocols` match and that we'll continue selection instead of rejecting connection), and `expect_source_type_match` should really be `expect_source_type_test` (meaning that source type was either `LOCAL` or `EXTERNAL` and we expect a test).